### DISCUSSION TITLE
Tips on how to get better stack trace

### DISCUSSION THREAD

**Discussion by sasq64** [mruby/mruby#5477] [Q&A] [answered]
Some of my code does not have debug-info; source locations and stracktrace shows `unknown`.
What do I need to think of to make this work ?

**Comment by matz** [comment, marked as answer]
The easiest way is adding `conf.enable_debug` in your build config file.

**Reply by sasq64** [reply]
I already have that, and some of the scripts I load have that info, but not all of them, and I haven't figured out what the difference is...

**Reply by matz** [reply]
If you are compiling your script using `mrbc`, add `-g` option.

**Reply by sasq64** [reply]
So it seems to happen for any code that I load dynamically, with for instance `mrb_load_file()` ...

**Reply by sasq64** [reply]
Aha, I needed to create an `mrbc_context` with the right settings!

**Reply by matz** [reply]
Can you show your `mrbc_context` setup code for other audiences?

**Comment by sasq64** [comment]
Sure;
```c
FILE* fp = fopen(name, "rbe");
if (fp != nullptr) {
    auto* ctx = mrbc_context_new(mrb);
    ctx->capture_errors = true;
    mrbc_filename(mrb, ctx, name);
    ctx->lineno = 1;
    mrb_load_file_cxt(mrb, fp, ctx);
    mrbc_context_free(mrb, ctx);
    fclose(fp);
}
```
